.github/workflows/ci-extends.yml

```diff
@@ -0,0 +1,41 @@
+---
+
+name: ci-extends
+
+on:
+  push:
+    branches:
+      - main
+      - feature/github-actions
+  pull_request:
+    branches:
+      - main
```

Comment:
These event will fail on forked PRs when accessing later on the secrets, what's the main goal for this particular workflow?

Reply:
The initial goal was to separate the workflow with and without secrets. We still need to validate that the docker image is buildable and pushable.

Reply:
I'm not fully onboarded to use this workflow on PRs for vary reasons:

- it won't work for forked PRs.
- security concerns, infosec already pointed us that random code should be approved by Elasticians before interacting with any GH secrets.

On the other hand, can we run this only for the branches in the upstream? If so, what do you think a different workflow name something like `publish-docker`?

If needed we can enable the Support for PRs in a follow up, but I prefer to play safe in terms of using secrets

Reply:
I split the workflow into two sub-steps. We will build the docker image within the `ci` workflow without secrets to validate that we can do so. On the other one `ci-extends`, we create and publish the docker image to the registry only on push events.